language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -242,4 +278,207 @@ private static Token getDefaultTokenToLeftOrRight(TokenStream tokenStream, int s
         }
         return token;
     }
+
+    /**
+     * Generate json representation for the given node.
+     * @param node                          Node to get the json representation
+     * @param anonStructs                   Map of anonymous structs
+     * @return {@link JsonElement}          Json Representation of the node
+     */
+    public static JsonElement generateJSON(Node node, Map<String, Node> anonStructs) {
+        if (node == null) {
+            return JsonNull.INSTANCE;
+        }
+        Set<Method> methods = ClassUtils.getAllInterfaces(node.getClass()).stream()
+                .flatMap(aClass -> Arrays.stream(aClass.getMethods()))
+                .collect(Collectors.toSet());
+        JsonObject nodeJson = new JsonObject();
+
+        JsonArray wsJsonArray = new JsonArray();
+        Set<Whitespace> ws = node.getWS();
+        if (ws != null && !ws.isEmpty()) {
+            for (Whitespace whitespace : ws) {
+                JsonObject wsJson = new JsonObject();
+                wsJson.addProperty("ws", whitespace.getWs());
+                wsJson.addProperty("i", whitespace.getIndex());
+                wsJson.addProperty("text", whitespace.getPrevious());
+                wsJson.addProperty("static", whitespace.isStatic());
+                wsJsonArray.add(wsJson);
+            }
+            nodeJson.add("ws", wsJsonArray);
+        }
+        org.ballerinalang.util.diagnostic.Diagnostic.DiagnosticPosition position = node.getPosition();
+        if (position != null) {
+            JsonObject positionJson = new JsonObject();
+            positionJson.addProperty("startColumn", position.getStartColumn());
+            positionJson.addProperty("startLine", position.getStartLine());
+            positionJson.addProperty("endColumn", position.getEndColumn());
+            positionJson.addProperty("endLine", position.getEndLine());
+            nodeJson.add("position", positionJson);
+        }
+
+        /* Virtual props */
```

Comment:
do we need this?

Reply:
Fixed with 69429412db1acf48667f376ed341a05af541dd3a